src/frontend/src/observer/observer_manager.rs

```diff
@@ -126,7 +126,7 @@ impl ObserverNodeImpl for FrontendObserverNode {
             }
         }
         catalog_guard.set_version(resp.version);
-        self.catalog_updated_tx.send(resp.version).unwrap();
```

Comment:
🤔, what if during ddl request re-subscribe occur?  The ddl request might hang if lose version notify.

Reply:
> 🤔, what if during ddl request re-subscribe occur? The ddl request might hang if lose version notify.

ddl request's version notify should be sent by handle_catalog_notification.

Reply:
Supposed the catalog version before meta reboots is 100. Here we will get `resp.version` with value 1. If the ddl request needs this version, there may be something wrong.

Reply:
> ddl request's version should be sent by `handle_catalog_notification`.

I mean the notification might be lost due to broken connection, and then frontend re-subscribe with full snapshot. But the dll request is still wait the version and will not be notified. The request will hang.

Reply:
> > ddl request's version should be sent by `handle_catalog_notification`.
> 
> I mean the notification might be lost due to broken connection, and then frontend re-subscribe with full snapshot. But the dll request is still wait the version and will not be notified. The request will hang.

If the stream is broken due to network jitter but the dml rpc returns successfully, the request will hang. I will add a commit to fix it.

Reply:
> Supposed the catalog version before meta reboots is 100. Here we will get `resp.version` with value 1. If the ddl request needs this version, there may be something wrong.

The ddl request do need a version, this's by design. We should figure out a solution to cover this case as well.

Reply:
> The ddl request do need a version, this's by design. We should figure out a solution to cover this case as well.

We may need to save the notification version in meta_store and restore it after meta reboots.

Reply:
> > > ddl request's version should be sent by `handle_catalog_notification`.
> > 
> > 
> > I mean the notification might be lost due to broken connection, and then frontend re-subscribe with full snapshot. But the dll request is still wait the version and will not be notified. The request will hang.
> 
> If the stream is broken due to network jitter but the ddl rpc returns successfully, the request will hang. I will add a commit to withdraw this modification.

In this situation, if meta reboots before re-subscribe, frontend will receive a version with `1` and the ddl request will wait until the new version is bigger than the value before meta reboots. So we may need to save the notification version in meta_store and restore it after meta reboots.

Reply:
Suggest to keep a `FIXME` here and rise a issue. Actually we might do some refactor in notification service after we de-couple ckpt and barrier.